DesiMock

```diff
@@ -0,0 +1,195 @@
+        self.prediction = Prediction(loc_pred=loc_pred, loc_conf=loc_conf, offsets=offsets, density_data=density_data_flat)
+        # Peaks
+        _ = compute_peaks(self)
+        # Absorbers?
+        add_abs_to_sightline(self)
+
+
+class DesiMock:
+    """
+    a class to load all spectrums from a Desi fits file.
+    :attribute wavelength array-like, save the wavelength of all spectrum (all spectrum have same wavelength array)
+    :attribute data, dict, using each spectra's id as its key and a dict of all data we need of this spectra as its value
+      its format like spectra_id: {'FLUX':flux, 'ERROR':error, 'z_qso':z_qso, 'RA':ra, 'DEC':dec, 'DLAS':dlas'information}
+      I think this format may be more readable and practical than simply take the data of each spectra apart.
+    """
+
+    def _init_(self, wavelength = None, data = {}):
+        self.wavelength = wavelength
+        self.data = data
+
+
+    def read_fits_file(self, spec_path, truth_path, zbest_path):
+        """
+        read Desi Mock spectrum from a fits file, load all spectrum as a DesiMock object
+        :param spec_path: spectrum file path
+        :param truth_path: truth file path
+        :param zbest_path: zbest file path
+        :return: self.wavelength,self.data(contained all information we need)
+        """
+        spec = fits.open(spec_path)
+        truth = fits.open(truth_path)
+        zbest = fits.open(zbest_path)
+
+        self.wavelength = np.hstack((spec[2].data.copy(), spec[7].data.copy(), spec[12].data.copy()))
+
+        dlas_data = truth[3].data[truth[3].data.copy()['NHI']>19.3]
+        spec_dlas = {}
+        for item in dlas_data:
+            if item[2] not in spec_dlas:
+                spec_dlas[item[2]] = [('00'+str(item[3]-item[2]*1000),item[1],item[0])] # (dla_id, dla)
+            else:
+                spec_dlas[item[2]].append(('00'+str(item[3]-item[2]*1000),item[1],item[0]))
+
+        test = np.array([True if item in dlas_data['TARGETID'] else False for item in spec[1].data['TARGETID'].copy()])
+        for item in spec[1].data['TARGETID'].copy()[~test]:
+            spec_dlas[item] = []
+
+        spec_id = spec[1].data['TARGETID'].copy()
+        flux1 = spec[3].data.copy()
+        flux2 = spec[8].data.copy()
+        flux3 = spec[13].data.copy()
+        flux = np.hstack((flux1,flux2,flux3))
+        ivar1 = spec[4].data.copy()
+        ivar2 = spec[9].data.copy()
+        ivar3 = spec[14].data.copy()
+        error = 1./np.sqrt(np.hstack((ivar1,ivar2,ivar3)))
+        z_qso = zbest[1].data['Z'].copy()
+        ra = spec[1].data['TARGET_RA'].copy()
+        dec = spec[1].data['TARGET_DEC'].copy()
+
+        self.data = {spec_id[i]:{'FLUX':flux[i],'ERROR': error[i], 'z_qso':z_qso[i] , 'RA': ra[i], 'DEC':dec[i], 'DLAS':spec_dlas[spec_id[i]]} for i in range(len(spec_id))}
+
+        return self.wavelength,self.data
+
+    def read_sightline(self,id):
```

Comment:
how about `get_sightline` instead of `read_sightline` ?

Reply:
I think ‘get_sightline’  is better than 'read_sightline' since 'read_sightline' means we get data from a file. I have changed it, thanks!